File: programmers/그리디/큰 수 만들기.py
def solution(number, k):
    answer = ''
    used = [True] * len(number)
    print(used)
    for n in range(10):
        if k == 0:
            break
        for i in range(len(number)):
            print(i)
            if k == 0:
                break
            if used[i] and number[i] == str(n):
                used[i] = False
                k -= 1
    for i in range(len(number)):
        if used[i]:
            answer += number[i]
    
    return answer

number = "1924"
k = 2
print(solution(number,k))

File: programmers/그리디/큰 수 만들기.java
import java.util.*;

class Solution {
    public String solution(String number, int k) {
        String answer = "";
        Stack<String> ans = new Stack<>();
        for (int i = 0; i < number.length(); i++){
            if (ans.isEmpty()){
                ans.push(Character.toString(number.charAt(i)));
                continue;
            }
            if (k > 0){
                while (!ans.isEmpty() && Integer.parseInt(ans.peek()) < Integer.parseInt(Character.toString(number.charAt(i)))){
                    if (k < 1) break;
                    ans.pop();
                    k --;
                }
            }
            ans.push(Character.toString(number.charAt(i)));
        }
        while (!ans.isEmpty()){
            String n = ans.pop();
            if (k > 0){
                k --;
                continue;
            }
            answer+=n;
        }
        StringBuffer sb = new StringBuffer(answer);
        answer = sb.reverse().toString();
        return answer;
    }
}